src/viam/app/data/client.py

```diff
@@ -290,10 +328,226 @@ async def bounding_box_labels_by_filter(self, filter: Optional[Filter] = None) -
-    async def data_capture_upload(self, metadata: UploadMetadata, sensor_contents: Optional[List[SensorData]] = None) -> None:
-        raise NotImplementedError()
+    # TODO [DATA-1663]: Fix Timestamps bug.
+    async def binary_data_capture_upload(
+        self,
+        part_id: str,
+        component_type: str,
+        component_name: str,
+        method_name: str,
+        method_parameters: Optional[Mapping[str, Any]],
+        file_extension: Optional[str],
+        tags: Optional[List[str]],
+        timestamps: Optional[tuple[Timestamp, Timestamp]],
+        binary_data: bytes
+    ) -> None:
+        """Upload binary sensor data.
+
+        Sync binary data collected on a robot through a specific component (i.e., a motor) along with the relevant metadata with
+        app.viam.com. Binary data can be found under the "Files" tab in Data on app.viam.com.
 
-    # TODO(RSDK-3637): Implement
-    async def file_upload(self, metadata: UploadMetadata, file_contents: Optional[FileData]) -> None:
-        raise NotImplementedError()
+        Args:
+            part_id (str): Part ID of the component used to captuer the data.
+            component_type (str): Type of the component used to capture the data (i.e., "movement_sensor").
+            component_name (str): Name of the component used to capture the data.
+            method_name (str): Name of the method used to capture the data.
+            method_parameters (Optional[Mapping[str, Any]]): Optional dictionary of method parameters. No longer in active use.
+            file_extension (Optional[str]): Optional file extension of data being uploaded.
+            tags (Optional[List[str]]): (Optional) Optional list of tags to allow for tag-based data filtering.
+            timestamps (Optional[tuple[google.protobuf.timestamp_pb2.Timestamp, google.protobuf.timestamp_pb2.Timestamp]]): Optional tuple
+                containing Timestamps denoting the times this data was requested[0] and received[1] by the appropriate sensor.
+            binary_data (bytes): The data to be uploaded, respresented in bytes.
+
+        Raises:
+            GRPCError: If an invalid part ID is passed.
+        """
+        sensor_contents = SensorData(
+            metadata=SensorMetadata(
+                time_requested=timestamps[0] if timestamps[0] else None,
+                time_received=timestamps[1] if timestamps[1] else None,
+            ),
+            struct=None,  # Used for tabular data.
+            binary=binary_data
+        )
+        metadata = UploadMetadata(
+            part_id=part_id,
+            component_type=component_type,
+            component_name=component_name,
+            method_name=method_name,
+            type=DataType.DATA_TYPE_BINARY_SENSOR,
+            file_name=None,  # Not used in app.
+            method_parameters=method_parameters if method_parameters else None,
+            file_extension=file_extension if file_extension else None,
```

Comment:
file_name is none because it doesn't seem to be used at all on the receiving end in app (i.e., it's not stored in app), but I left it in explicitly defined as None as a reminder that it is a parameter that exists in case something changes in the future. I left file_extension in as an option just because it is used in app, but I think we could totally get away with not including it. In app, really no processing is done on the file extension, but it is stored as meta data if it nonempty. What do you think?

Reply:
It seems really strange to include a file extension but not a file name, so I'm inclined to not include either. If app starts caring about it, we can add it back as an optional param with default without breaking anything.